### PR TITLE
Added compatibility with uint8 to SSIM metric

### PR DESCRIPTION
**Pull request by MarcBresson**
Add compatibility and tests for when y_pred and y are in uint8 dtype

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/ssim.py [hunk 1]
```diff
@@ -157,6 +159,15 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
                 f"Expected y_pred and y to have BxCxHxW shape. Got y_pred: {y_pred.shape} and y: {y.shape}."
             )
 
+        if y_pred.dtype == torch.uint8 or y.dtype == torch.uint8:
+            if self.data_range != 255:
+                warnings.warn(
+                    "dtypes of the input tensors are torch.uint8 but data range is not set to 255.", RuntimeWarning
+                )
```
Comment (vfdev-5) [inline ignite/metrics/ssim.py:166]: I'm not a big fan of showing a warning in this case. I would follow the same strategy as skimage where they just transform to floating point. So, I would do the following:
```
if not y.is_floating_point():
    y = y.float()
if not y_pred.is_floating_point():
    y_pred = y_pred.float()
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for working on this PR @MarcBresson !
I left a comment on how I would tackle this dtype problem.
I also label this PR with "help wanted" label to create a thread on our discord, where we could discuss about this PR in a more fluent way

File: ignite/metrics/ssim.py (hunk 1 above)
Comment (MarcBresson) [reply ignite/metrics/ssim.py:166]: Maybe we should put a default value to the data_range argument too? Most of the time we are dealing with [0; 1] images.

Comment (vfdev-5) [reply ignite/metrics/ssim.py:166]: I do not have a strong opinion on this. We have to figure out if there wont be any negative impact for the users in this case.

Comment (MarcBresson) [reply ignite/metrics/ssim.py:164]: I think the skiimg approach is quite good: default to data_range = 1.0, and warn if the detected type is of integer and that data range is still set to 1

**Comment by vfdev-5** [comment]
@MarcBresson can you please resolve the conflict and let's apply what we discussed up to here: https://github.com/pytorch/ignite/pull/3045#discussion_r1305430207
Concerning default to data_range = 1.0, let's keep the public API the same for now.

**Review by vfdev-5 — APPROVED** [review]
Thanks @MarcBresson , LGTM!

**Comment by MarcBresson** [comment]
Sorry for the delay btw! Got caught up in a job seeking activity ahah

**Comment by vfdev-5** [comment]
No worries, hope you can get something exciting as a new job role!

**Pull request merged**